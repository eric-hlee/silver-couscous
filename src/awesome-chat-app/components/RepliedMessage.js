import React from 'react'

class RepliedMessage extends React.Component {

  render() {

    const style_messageBubbleGroup = {
      position: 'relative',
      overflow: 'auto',
    }

    const style_repliedMessageBubble = {
      display: 'inline-block',
      border: '1px solid #555555',
      borderRadius: '10px',
      padding: '15px',
      marginBottom: '15px'
    }

    const style_repliedMessage = {
      marginTop: '6px',
      fontFamily: 'Helvetica Neue, Arial, sans-serif',
      fontSize: '16px'
    }
    
    return (
      <div className="MessageBubbleGroup" style={style_messageBubbleGroup}>
        <div className="RepliedMessageBubble" style={style_repliedMessageBubble}>
          <div className="Date">
            <span>Sent on {this.props.date.toString()}</span>
          </div>
          <div className="Message" style={style_repliedMessage}>
            <span>{this.props.text}</span>
          </div>
        </div>
      </div>
    );
  }
}

export default RepliedMessage