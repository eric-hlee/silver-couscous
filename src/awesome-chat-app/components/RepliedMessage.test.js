import React from 'react'
import { shallow } from 'enzyme'
import RepliedMessage from './RepliedMessage'

describe('RepliedMessage', () => {
  it('should render self and subcomponents', () => {
    const wrapper = shallow(<RepliedMessage key='' date='' text=''/>)
    expect(wrapper).toMatchSnapshot()
  })
})
