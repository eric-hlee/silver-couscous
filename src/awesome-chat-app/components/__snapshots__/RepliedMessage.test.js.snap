// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepliedMessage should render self and subcomponents 1`] = `
<div
  className="MessageBubbleGroup"
  style={
    Object {
      "overflow": "auto",
      "position": "relative",
    }
  }
>
  <div
    className="RepliedMessageBubble"
    style={
      Object {
        "border": "1px solid #555555",
        "borderRadius": "10px",
        "display": "inline-block",
        "marginBottom": "15px",
        "padding": "15px",
      }
    }
  >
    <div
      className="Date"
    >
      <span>
        Sent on 
      </span>
    </div>
    <div
      className="Message"
      style={
        Object {
          "fontFamily": "Helvetica Neue, Arial, sans-serif",
          "fontSize": "16px",
          "marginTop": "6px",
        }
      }
    >
      <span />
    </div>
  </div>
</div>
`;
